tests/authentication/04-check-keys-remote.t

```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+# THIS FILE IS PART OF THE CYLC SUITE ENGINE.
+# Copyright (C) NIWA & British Crown (Met Office) & Contributors.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#-------------------------------------------------------------------------------
+# Checks remote ZMQ keys are created and deleted on shutdown.
+. "$(dirname "$0")/test_header"
+set_test_remote_host
+set_test_number 4
+init_suite "${TEST_NAME_BASE}" <<'__SUITE_RC__'
+[cylc]
+[scheduling]
+    [[graph]]
+        R1 = holder => held
+[runtime]
+    [[holder]]    
+        script = """cylc hold "${CYLC_SUITE_NAME}" """
+        [[[remote]]]
+            host = $CYLC_TEST_HOST
+__SUITE_RC__
+
+run_ok "${TEST_NAME_BASE}-validate" cylc validate "${SUITE_NAME}" 
+suite_run_ok "${TEST_NAME_BASE}-run" cylc run "${SUITE_NAME}" 
+RRUND="cylc-run/${SUITE_NAME}"
+RSRVD="${RRUND}/.service"
+poll_grep_suite_log 'Holding all waiting or queued tasks now'
+SSH='ssh -n -oBatchMode=yes -oConnectTimeout=5'
+${SSH} "${CYLC_TEST_HOST}" \
+find "${RSRVD}" -type f -name "*key*"|awk -F/ '{print $NF}'|sort >'find.out'
+cmp_ok 'find.out' <<'__OUT__'
+client.key
+client.key_secret
+server.key
+__OUT__
+cylc stop --max-polls=60 --interval=1 "${SUITE_NAME}"
+${SSH} "${CYLC_TEST_HOST}"
```

Comment:
Is this line meant to be here? Or is it missing a trailing `\`?

Reply:
So this is probably highlighting my inexperience. I guess I wasn't exactly sure when commands are executed on remote host or on scheduler, and I wanted to make sure that the find command would be exectued on the remote host. So, maybe I don't really need the ssh if cylc stop doesn't pull me back to the scheduler? The tests pass either way for me!

Reply:
(So the entire test script, like the scheduler program, runs on the suite host).